src/supplemental/nanolib/parquet/parquet.cc

```diff
@@ -490,6 +484,46 @@ parquet_write(
 			goto again;
 	}
 
+	char md5_buffer[33];
+	int ret = Compute_file_md5(filename, md5_buffer);
+	if (ret != 0) {
+		log_error("Failed to calculate md5sum");
+		free(filename);
```

Comment:
need to pthread_mutex_unlock(&parquet_queue_mutex);

Reply:
thx, done